### PR TITLE
Save pullid,branch,parent_commit_id on commit if None

### PR DESCRIPTION
**Pull request by thomasrockhu-codecov**
## Problem

Users are seeing `Missing head report` error in the Codecov UI, despite uploading coverage for a commit that should be the head report. Looking into this, sometimes a PR on a given repo WILL have the error, sometimes it won't. Fix here is that the FIRST upload that comes in determines whether or not we set a `pullid` for a commit. Subsequent uploads will not change this.

This PR changes this by setting `pullid`, `branch`, `parent_commit_id` to the most recent value that is not `nil`. This ensures that future uploads can set these values if necessary.

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

---

> [!NOTE]
> Populate `branch`, `parent_commit_id`, and `pullid` on existing commits if missing during upload and call `update_commit` when the commit is created or updated, with tests added.
> 
> - **API/Serializers**
>   - Update `CommitSerializer.create` to backfill `branch`, `parent_commit_id`, and `pullid` on existing commits only when those fields are `None`.
>   - Persist changes to the commit if any field was updated and always send `upload_breadcrumb` for existing commits.
>   - Invoke `TaskService.update_commit` when the commit is newly created or updated (previously only on create).
> - **Tests**
>   - Add tests to confirm fields are set when initially `None` and not overwritten when already present during `upload_coverage`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 55756bf8d852e83a6648fd0d4c50de7b1a559364. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: apps/codecov-api/upload/serializers.py [hunk 1]
```diff
@@ -159,6 +170,10 @@ def create(self, validated_data: dict[str, Any]) -> Commit:
                 ),
             )
 
+        if created or updated:
```
[the remark this answers is not in the document]
Comment (thomasrockhu-codecov) [reply apps/codecov-api/upload/serializers.py]: I think this is fine after the breadcrumb as this task is doing `Branch` and `Pull` object updates.

**Review by adrianviquez — APPROVED** [review]
